security/access_denied_handler.rst

```diff
@@ -0,0 +1,68 @@
+.. index::
+    single: Security; Creating a Custom Access Denied Handler
+
+How to Create a Custom Access Denied Handler
+============================================
+
+When your application throw an ``AccessDeniedException`` you can catch this exception
+with a service to return a custom Response.
+
+On each firewall context you can define a custom access denied handler.
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # app/config/security.yml
+        firewalls:
+            foo:
+                # ...
+                access_denied_handler: custom_handler.service.id
+
+    .. code-block:: php
+
+        // app/config/security.php
+        $container->loadFromExtension('security', array(
+            'firewalls' => array(
+                'foo' => array(
+                    // ...
+                    'access_denied_handler' => 'custom_handler.service.id',
+                ),
+            ),
+        ));
+
+
+Your handler must implement the interface
+:class:`Symfony\\Component\\Security\\Http\\Authorization\\AccessDeniedHandlerInterface`.
+This interface define one method called ``handle()`` that can do whatever you want.
+You can use it to send a mail, log a message, or generally return a custom Response.
+
+
+.. code-block:: php
+
+    namespace AppBundle\Security;
```

Comment:
What would be a good place for this class ? For me `AppBundle\Security` look like a perfect default place for a `Foo` project.